src/agentscope/rag/rag.py

```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+"""
+Base class module for retrieval augmented generation (RAG).
+To accommodate the RAG process of different packages,
+we abstract the RAG process into four stages:
+- data loading: loading data into memory for following processing;
+- data indexing and storage: document chunking, embedding generation,
+and off-load the data into VDB;
+- data retrieval: taking a query and return a batch of documents or
+document chunks;
+- post-processing of the retrieved data: use the retrieved data to
+generate an answer.
+"""
+
+from abc import ABC, abstractmethod
+from typing import Any, Optional
+
+from agentscope.models import ModelWrapperBase
+
+DEFAULT_CHUNK_SIZE = 1024
+DEFAULT_CHUNK_OVERLAP = 20
+DEFAULT_TOP_K = 5
+
+
+class RAGBase(ABC):
+    """
+    Base class for RAG, CANNOT be instantiated directly
+    """
+
+    def __init__(
+        self,
+        model: Optional[ModelWrapperBase],
+        emb_model: Any = None,
```

Comment:
why the type of `emb_model` is Any?

Reply:
Because as a base model, its children class can accept our embedding model wrapper and embedding model from LlamaIndex or Langchain.